openvdb_houdini/houdini/PointUtils.cc

```diff
@@ -1619,154 +1619,166 @@ attributeStorageType(const GA_Attribute* const attribute)
+        const openvdb::Name& codecType = type.second;
 
-            const Name valueType = type.first;
-            const Name codecType = type.second;
+        os << it.first << "[" << type.first;
 
-            infoStr << it->first << "[";
+        // if no value compression, hide the codec
+        os << (codecType != "null" ? "_" + codecType : "");
+        os << "]";
+        if (!array.hasConstantStride()) os << " [dynamic]";
+        else if (array.stride() > 1) os << " [" << array.stride() << "]";
+    }
 
-            // if no value compression, hide the codec from the middle-click output
+    attributeStr = (os.str().empty() ? "none" : os.str());
+}
 
-            if (codecType == "null") {
-                infoStr << valueType;
-            }
-            else if (isString(array)) {
-                infoStr << "str";
-            }
-            else {
-                infoStr << valueType << "_" << codecType;
-            }
+void
+pointDataGridSpecificInfoText(std::ostream& infoStr, const GridBase& grid)
+{
+    const PointDataGrid* pointDataGrid =
+        dynamic_cast<const PointDataGrid*>(&grid);
+    if (!pointDataGrid) return;
 
-            if (!array.hasConstantStride()) {
-                infoStr << "[dynamic]";
-            }
-            else if (array.stride() > 1) {
-                infoStr << "[" << array.stride() << "]";
-            }
+    // match native OpenVDB convention as much as possible
 
-            infoStr << "]";
-        }
+    infoStr << " voxel size: " << pointDataGrid->transform().voxelSize()[0] << ",";
+    infoStr << " type: points,";
+
+    if (pointDataGrid->activeVoxelCount() != 0) {
+        const Coord dim = grid.evalActiveVoxelDim();
+        infoStr << " dim: " << dim[0] << "x" << dim[1] << "x" << dim[2] << ",";
+    } else {
+        infoStr <<" <empty>,";
+    }
+
+    std::string countStr, groupStr, attributeStr;
+    collectPointInfo(*pointDataGrid, countStr, groupStr, attributeStr);
+
+    infoStr << " count: " << countStr << ",";
+    infoStr << " groups: " << groupStr << ",";
+    infoStr << " attributes: " << attributeStr;
+}
+
+void
+populateInfoTree(UT_InfoTree& infoTree, const GU_Detail& geo) {
+
+    infoTree.addColumnHeading("Count");
+    infoTree.addColumnHeading("Groups");
+    infoTree.addColumnHeading("Attributes");
+
+    for (VdbPrimCIterator it(&geo); it; ++it) {
+        const openvdb::GridBase::ConstPtr grid = it->getConstGridPtr();
+        if (!grid) continue;
+        if (!grid->isType<openvdb::points::PointDataGrid>()) continue;
+
+        const PointDataGrid& points = *gridConstPtrCast<PointDataGrid>(grid);
+
+        std::string countStr, groupStr, attributeStr;
+        collectPointInfo(points, countStr, groupStr, attributeStr);
 
-        if (first)  infoStr << "<none>";
+        ut_PropertyRow* row = infoTree.addProperties();
+        row->append(countStr);
+        row->append(groupStr);
+        row->append(attributeStr);
```

Comment:
I'm curious how SideFX are using this logic. The "Sparse Volumes" branch that contains "Point Count", "Point Attributes", etc also contains other entries - "Primitive", "Name", "Display Mode", etc. It looks like this function is hard-coded to only have three columns per row, so I'm assuming this logic must have been pulled across to another implementation somewhere and this function isn't used directly? Is it perhaps in SOP_Node::fillInfoTreeNodeSpecific?

Reply:
We are using a version of collectPointInfo that is called from GU_DetailInfo::fillInfoTree.  mako then finds the relevant tables from that, so this is how the three columns are passed over.  All the code in populateInfoTree isn't used for the generic GU_DetailInfo as that is outside of the per-node world.  There isn't a way right now to register a generic infotree on geometry; only on nodes; which is why the bespoke stuff here only shows up on the OpenVDB nodes and not the other nodes.